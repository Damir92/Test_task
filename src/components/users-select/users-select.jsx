import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {FormControl, InputLabel, Select, MenuItem, makeStyles} from '@material-ui/core';

import {getUserGroups, getGroup} from '../../store/selectors';
import {setGroup} from '../../store/action';

const useStyles = makeStyles((theme) => ({
  formControl: {
    display: `flex`,
    justifyContent: `center`,
    marginTop: 50,
    marginBottom: 50,
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
}));

const UsersSelect = ({groups, activeGroup, onGroupChange}) => {
  const classes = useStyles();

  const handleChange = (evt) => {
    onGroupChange(evt.target.value);
  };

  return (
    <FormControl className={classes.formControl}>
      <InputLabel id="demo-simple-select-label">Group</InputLabel>
      <Select
        labelId="demo-simple-select-label"
        id="demo-simple-select"
        value={activeGroup}
        onChange={handleChange}
      >
        <MenuItem value=''>None</MenuItem>
        {groups.map((item) => (
          <MenuItem key={item} value={item} name="group">{item}</MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

UsersSelect.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeGroup: PropTypes.string.isRequired,
  onGroupChange: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  groups: getUserGroups(state),
  activeGroup: getGroup(state),
});

const mapDispatchToProps = (dispatch) => ({
  onGroupChange: (data) => dispatch(setGroup(data)),
});

export {UsersSelect};
export default connect(mapStateToProps, mapDispatchToProps)(UsersSelect);
