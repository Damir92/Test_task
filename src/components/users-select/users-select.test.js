import React from 'react';
import {shallow} from 'enzyme';

import {UsersSelect} from './users-select';

const groups = [`user`, `admin`, `director`];


describe(`Users select component`, () => {
  it(`UsersSelect snapshot`, () => {
    const tree = shallow(
        <UsersSelect
          groups={groups}
          activeGroup={``}
          onGroupChange={() => {}}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
