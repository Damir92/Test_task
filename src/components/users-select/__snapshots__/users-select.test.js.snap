// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users select component UsersSelect snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
>
  <WithStyles(ForwardRef(InputLabel))
    id="demo-simple-select-label"
  >
    Group
  </WithStyles(ForwardRef(InputLabel))>
  <WithStyles(ForwardRef(Select))
    id="demo-simple-select"
    labelId="demo-simple-select-label"
    onChange={[Function]}
    value=""
  >
    <WithStyles(ForwardRef(MenuItem))
      value=""
    >
      None
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="user"
      name="group"
      value="user"
    >
      user
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="admin"
      name="group"
      value="admin"
    >
      admin
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="director"
      name="group"
      value="director"
    >
      director
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</WithStyles(ForwardRef(FormControl))>
`;
